Security Chapter - Simple Auth and some restructure (#123)
hazelcast/hz-docs: docs/modules/security/pages/simple-authentication.adoc

= Simple authentication
[[simple-authentication]]

The simple authentication enables you to define users and their roles **directly** in the
Hazelcast member configuration.

The xref:default-authentication.adoc[default authentication] is based on the member's
identity configuration (when defined) or cluster name (otherwise); it does not
allow defining users and assigning them roles.

> And when using the xref:jaas-authentication.adoc[JAAS-based authentication], you need
> to provide your login modules themselves or set of enterprise level authentication implementations
> (LDAP, Kerberos, etc.) which require additional infrastructure to have them working.
> 
> Simple authentication closes the gap between the default authentication and
> advanced JAAS-based authentication methods.

Reviewer: I would try to keep mentioning JAAS as "the last resort". Something like:

```suggestion
And when using the advanced authentication methods, you either need additional infrastructure for Hazelcast's enterprise-level authentication (LDAP server, Kerberos, etc.) or you need to provide your login module implementations in xref:jaas-authentication.adoc[JAAS-based authentication].

Simple authentication closes the gap between the default authentication and
advanced authentication methods.
```

> On the member side it'll use already existing SimplePropertiesLoginModule implementation.

Reviewer: I wouldn't include this paragraph. It is an implementation detail.
